common/src/main/java/net/william278/husktowns/hook/Pl3xMapHook.java

```diff
@@ -0,0 +1,147 @@
+package net.william278.husktowns.hook;
+
+import net.pl3x.map.core.Pl3xMap;
+import net.pl3x.map.core.event.EventHandler;
+import net.pl3x.map.core.event.EventListener;
+import net.pl3x.map.core.event.server.Pl3xMapEnabledEvent;
+import net.pl3x.map.core.event.world.WorldLoadedEvent;
+import net.pl3x.map.core.event.world.WorldUnloadedEvent;
+import net.pl3x.map.core.markers.Point;
+import net.pl3x.map.core.markers.layer.SimpleLayer;
+import net.pl3x.map.core.markers.marker.Marker;
+import net.pl3x.map.core.markers.option.Options;
+import net.pl3x.map.core.markers.option.Tooltip;
+import net.william278.husktowns.HuskTowns;
+import net.william278.husktowns.claim.TownClaim;
+import net.william278.husktowns.claim.World;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.logging.Level;
+import java.util.stream.Collectors;
+
+public class Pl3xMapHook extends MapHook implements EventListener {
+
+    private static final String CLAIMS_LAYER = "claim_markers";
+    private final ConcurrentLinkedQueue<TownClaim> claims = new ConcurrentLinkedQueue<>();
+
+    public Pl3xMapHook(@NotNull HuskTowns plugin) {
+        super(plugin, "Pl3xMap");
+    }
+
+    @Override
+    public void onEnable() {
+        Pl3xMap.api().getEventRegistry().register(this);
+
+        // TODO: Pl3xMap's enable and world load events are currently being ran before this method meaning that the claims layer isn't loading on server startup
+        if (Pl3xMap.api().isEnabled()) Pl3xMap.api().getWorldRegistry().forEach(this::registerLayers);
+
+        plugin.log(Level.INFO, "Enabled Pl3xMap markers hook. Populating web map with claims...");
+        for (World world : plugin.getWorlds()) {
+            plugin.getClaimWorld(world).ifPresent(claimWorld -> setClaimMarkers(claimWorld.getClaims(plugin), world));
+        }
+    }
+
+    @Override
+    public void setClaimMarker(@NotNull TownClaim claim, @NotNull World world) {
+        claims.add(claim);
```

Comment:
So for this (and methods below) what you'll want to do is get the claim world for the world if it exists (`#getClaimWorld(World)` in the main class) and then add it to the ConcurrentMap of ClaimWorld>TownClaim

Reply:
Resolved in [624f586](https://github.com/WiIIiam278/HuskTowns/pull/303/commits/624f58658bb6397408843cb1c7cf439b9782809a)